src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PostMergeValidationRules/IsInvalidFieldRule.cs

```diff
@@ -42,12 +42,8 @@ public void Handle(SchemaEvent @event, CompositionContext context)
             var fieldArgumentValue = (string)isDirective.Arguments[Field].Value!;
             var fieldSelectionMapParser = new FieldSelectionMapParser(fieldArgumentValue);
             var fieldSelectionMap = fieldSelectionMapParser.Parse();
-            var inputTypeNode = sourceArgument.Type.ToTypeNode();
-            var inputTypeDefinition = schema.Types[sourceArgument.Type.AsTypeDefinition().Name];
-            var inputType = inputTypeNode.RewriteToType(inputTypeDefinition);
-            var outputTypeNode = sourceField.Type.ToTypeNode();
-            var outputTypeDefinition = schema.Types[sourceField.Type.AsTypeDefinition().Name];
-            var outputType = outputTypeNode.RewriteToType(outputTypeDefinition);
+            var inputType = sourceArgument.Type;
+            var outputType = sourceField.Type;
```

Comment:
Hmm. I don't think that this will work if the field is in another source schema, I think that it needs to reference the type in the composite schema. `RequireInvalidFieldsRule` (similar) has tests for these cross-schema references, but there are no such tests for this rule.